lib/graphql/execution/lookahead.rb

```diff
@@ -94,6 +94,43 @@ def selection(field_name, arguments: nil)
         end
       end
 
+      # Like {#selection}, but for all nodes.
+      # It returns a list of Lookaheads for all Selections
+      #
+      # If `arguments:` is provided, each provided key/value will be matched
+      # against the arguments in each selection. This method will filter the selections
+      # if any of the given `arguments:` do not match the given selection.
+      #
+      # @example getting the name of a selection
+      #   def articles(lookahead:)
+      #     next_lookaheads = lookahead.selections # => [#<GraphQL::Execution::Lookahead ...>, ...]
+      #     next_lookaheads.map(&:name) #=> [:full_content, :title]
+      #   end
+      #
+      # @param arguments [Hash] Arguments which must match in the selection
+      # @return [Array<GraphQL::Execution::Lookahead>]
+      def selections(arguments: nil)
+        @ast_nodes
+          .flat_map(&:selections)
+          .map(&:name)
+          .map { |name| selection(name, arguments: arguments) }
+          .select(&:selected?)
```

Comment:
I _think_ this will have a bug if two of the `@ast_nodes` have two of the same selections, for example: 

```
a { b }
a { b c }
```

In that case, `.selections` will return three items, one for the first `b`, another for the second `b`, and one for `c`. It should only return 2 in that case, right? 

~~Also, I noticed you added `.select(&:selected?`. Did you find that unselected selections were return there? I can't imagine~~, Oh, nevermind, I guess it's for when `arguments` doesn't match, is that right? 

Anyhow, how about adding a test case for that multiple-selection issue to make sure it's handled properly?